dartagnan/src/main/java/com/dat3m/dartagnan/wmm/utils/RelationRepository.java

```diff
@@ -157,18 +163,30 @@ private Relation getBasicRelation(String name){
             case IDDTRANS:
                 return getRelation(RelTrans.class, getRelation(IDD));
             case DATA:
-                return getRelation(RelIntersection.class, getRelation(IDDTRANS), getRelation(RW)).setName(DATA);
+                return getRelation(RelIntersection.class, 
+                		getRelation(IDDTRANS), 
+                		getRelation(RelCartesian.class, 
+                				FilterUnion.get(FilterBasic.get(Tag.READ), FilterBasic.get(Tag.RISCV.STCOND)), 
+                				FilterBasic.get(Tag.WRITE))
+                		).setName(DATA);
```

Comment:
Can't we just always intersect with `MxM` (also for the other dep-relations)?
I think the fact that `data` is a subset of`RxW` in most memory models is an emergent property rather than an enforced one.

Reply:
That won't work for `ctrl` because `VISIBLE` is not a subset of `MEMORY`.
However using `M*M` for `data` and `addr` seems to cause no problems in any litmus test so I think we can use it for those two.